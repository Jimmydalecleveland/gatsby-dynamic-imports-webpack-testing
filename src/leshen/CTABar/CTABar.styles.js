import styled from "@emotion/styled"
import { css } from "@emotion/core"

import { ColorDeterminer } from "../utils/helpers.styles"

export const Container = styled.div(
  ColorDeterminer,
  ({ theme, light }) => css`
    background-color: ${light
      ? theme.colors.light.lighter.hex
      : theme.colors.primary.base.hex};
    color: ${light
      ? theme.colors.primary.base.hex
      : theme.colors.primary.base.readable};
    border-top: 5px solid;
    width: 100%;
    z-index: 3;

    &.original {
      border: 0;
    }
  `
)

export const StickyBar = styled.div(
  ({ isVisible }) => css`
    transform: translateY(${isVisible ? "0" : "100%"});
    z-index: 3;
    position: fixed;
    opacity: ${isVisible ? 1 : 0};
    bottom: 0;
    width: 100%;
    transition: transform 0.5s cubic-bezier(1, 0.05, 0.17, 1.09),
      opacity 0.5s cubic-bezier(1, 0.05, 0.17, 1.09);
    padding: 0;
  `
)
