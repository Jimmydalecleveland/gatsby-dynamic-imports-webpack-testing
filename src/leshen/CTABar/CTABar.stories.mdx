import { Story, Meta, Preview, Props, Anchor } from '@storybook/addon-docs/blocks'

import Typography from '../Typography'
import ComponentProgress from '../ComponentProgress'
import CTABar from '.'
import StickyBar from './StickyBar'
import { ExampleWrapper, ExampleAction as Button } from './ExampleAction'

import Section from '../Section'
import Hero from '../Hero'
import SplitContent from '../SplitContent'
import Footer from '../Footer'

import clearlinkLogo from '../Footer/logo.svg'
import { FOOTER_NAV_ITEMS, DISCLAIMERS } from '../Footer/dummyData'
import { FLUID_IMAGE_DATA, FLUID_IMAGE_DATA_SPLIT } from '../Hero/dummyData'
import { FLUID_IMAGE_DATA as SPLIT_FLUID_IMAGE_DATA } from '../SplitContent/dummyData'

import {
  Wifi,
  Phone,
  Done,
  QuestionAnswer,
  ChevronRight,
} from 'emotion-icons/material'

<Meta title='Components|CTA Bar' component={CTABar} />

# CTA Bar

<Preview>
  <Story name="default">
    <>
      <CTABar>
        <ExampleWrapper>
          <Button variant="feature" color="light" leftIcon={<Phone/>} ghost>
            Call Now 1.888.888.8888
          </Button>
          <Button variant="feature" color="light" leftIcon={<Done/>} ghost>
            Order Online
          </Button>
        </ExampleWrapper>
      </CTABar>
    </>
  </Story>
</Preview>

## Props
<Props of={CTABar} />

## Light Variations
<Preview>
  <Story name="light with ghost buttons">
    <CTABar light>
        <ExampleWrapper>
          <Button variant="feature" color="primary" leftIcon={<Phone/>} ghost>
            Call Now 1.888.888.8888
          </Button>
          <Button variant="feature" color="primary" leftIcon={<Done/>} ghost>
            Order Online
          </Button>
        </ExampleWrapper>
    </CTABar>
  </Story>
</Preview>

<!-- 
<Preview>
  <Story name="light with default buttons">
    <CTABar light>
      <ExampleWrapper>
        <Button variant="feature" leftIcon={<Phone/>}>
          Call Now
        </Button>
        <Button variant="feature" leftIcon={<QuestionAnswer/>}>
          Chat with agent
        </Button>
        <Button variant="feature" leftIcon={<Done/>}>
          Check Availability
        </Button>
      </ExampleWrapper>
    </CTABar>
  </Story>
</Preview>

## Different amounts of actions
The CTABar will automatically center the children (actions) given to it, with a gap inbetween.

<Preview>
  <Story name="different amounts">
    <>
      <p>3 actions</p>
      <CTABar>
        <ExampleWrapper>
          <Button variant="feature" color="light" leftIcon={<Phone/>} ghost>
            Call Now: 888-888-8888
          </Button>
          <Button variant="feature" color="light" leftIcon={<QuestionAnswer/>} ghost>
            Chat with agent
          </Button>
          <Button variant="feature" color="light" leftIcon={<Done/>} ghost>
            Check Availability
          </Button>
        </ExampleWrapper>
      </CTABar>
      <p>2 actions</p>
        <CTABar>
          <ExampleWrapper>
            <Button variant="feature" color="light" leftIcon={<Phone/>} ghost>
              Call Now: 888-888-8888
            </Button>
            <Button variant="feature" color="light" leftIcon={<QuestionAnswer/>} ghost>
              Chat with agent
            </Button>
          </ExampleWrapper>
        </CTABar>
      <p>1 action</p>
      <CTABar>
        <Button variant="feature" color="light"leftIcon={<Phone/>} ghost>
          Call Now: 888-888-8888
        </Button>
      </CTABar>
    </>
  </Story>
</Preview>

-->
## With Sticky Functionality
<Preview>
  <Story name="with sticky functionality">
    <>
      <Hero image={FLUID_IMAGE_DATA}>
        <Typography variant="h1">
          Bundle DIRECTV Packages and CenturyLink
        </Typography>
        <Typography variant="h4" rendersAs="h2">
          DIRECTV SELECT All-included + CenturyLink Internet up to 20 Mbps
          starting at
        </Typography>
        <Button leftIcon={<Phone />} variant="hero">
          Call Now 1.888.888.8888
        </Button>
      </Hero>
      <StickyBar showOriginal>
        <ExampleWrapper>
          <Button variant="feature" color="light" leftIcon={<Phone/>} ghost>
            Call Now 1.888.888.8888
          </Button>
          <Button variant="feature" color="light" leftIcon={<Done/>} ghost>
            Order Online
          </Button>
        </ExampleWrapper>
      </StickyBar>
      <SplitContent
        image={SPLIT_FLUID_IMAGE_DATA}
        prominent
        heading="H2 standard size headline"
        mainContent={
          <>
            <Typography>
              There is a lot of exciting stuff going on in the stars above us that
              make astronomy so much fun. The truth is the universe is a
              constantly changing, moving, some would say &quot;living&quot; thing
              because you just never know what you are going to see on any given
              night of stargazing.
            </Typography>
            <Button primary>Submit</Button>
          </>
        }
        />
      <SplitContent
        image={SPLIT_FLUID_IMAGE_DATA}
        heading="H2 standard size headline"
        mainContent={
          <>
            <Typography>
              There is a lot of exciting stuff going on in the stars above us that
              make astronomy so much fun. The truth is the universe is a
              constantly changing, moving, some would say &quot;living&quot; thing
              because you just never know what you are going to see on any given
              night of stargazing.
            </Typography>
            <Button primary>Submit</Button>
          </>
        }
      />
      <Footer
        logo={clearlinkLogo}
        navData={FOOTER_NAV_ITEMS}
        disclaimers={DISCLAIMERS}
      />
    </>
  </Story>
</Preview> 
