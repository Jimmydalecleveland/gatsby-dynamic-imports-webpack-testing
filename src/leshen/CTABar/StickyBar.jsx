import React, { useEffect, useState, useRef } from 'react'
import PropTypes from 'prop-types'
import { Global, css } from '@emotion/core'
import useOnScroll from '../../hooks/useOnScroll'
import CTABar from '.'
import * as Styled from './CTABar.styles'

const StickyBar = ({ children, showOriginal }) => {
  const { scrollRef, scrolledTo } = useOnScroll()

  const stickyBarRef = useRef()
  const [stickyBarHeight, setStickyBarHeight] = useState(80)

  useEffect(() => {
    setStickyBarHeight(stickyBarRef.current.clientHeight)
  }, [stickyBarRef])

  return (
    <>
      <Global
        styles={css`
          body {
            margin-bottom: ${stickyBarHeight}px;
          }
        `}
      />
      {showOriginal && <CTABar className="original">{children}</CTABar>}
      <div id="sticky-ref" ref={scrollRef} />
      <Styled.StickyBar
        ref={stickyBarRef}
        id="sticky-bar"
        isVisible={scrolledTo}
      >
        <CTABar className="sticky">{children}</CTABar>
      </Styled.StickyBar>
    </>
  )
}

StickyBar.defaultProps = {
  showOriginal: false,
}

StickyBar.propTypes = {
  children: PropTypes.node.isRequired,
  showOriginal: PropTypes.bool,
}

export default StickyBar
