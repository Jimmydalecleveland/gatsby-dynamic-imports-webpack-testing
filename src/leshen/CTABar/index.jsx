import React from 'react'
import PropTypes from 'prop-types'

import * as Styled from './CTABar.styles'

const CTABar = ({ light, children, ...props }) => (
  <Styled.Container light={light} {...props}>
    {children}
  </Styled.Container>
)

CTABar.defaultProps = {
  light: false,
}

CTABar.propTypes = {
  /** Sets the colors to be the light theme, or dark if false */
  light: PropTypes.bool,
  children: PropTypes.node.isRequired,
}

export default CTABar

export const ticketID = 'CLCMS-203'
