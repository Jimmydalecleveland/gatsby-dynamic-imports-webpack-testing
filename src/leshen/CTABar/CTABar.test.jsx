import React from 'react'
import { render } from 'setup-tests'

import { Phone, Done, QuestionAnswer } from 'emotion-icons/material'
import CTABar from '.'
import StickyBar from './StickyBar'
import Button from '../Button'
import StandardImplementation from './standardImplementation'

jest.mock('emotion-icons/material', () => ({
  Phone: () => <div>Phone icon Mock</div>,
  Done: () => <div>Done icon Mock</div>,
  QuestionAnswer: () => <div>QuestionAnswer icon Mock</div>,
}))

describe('CTABar', () => {
  test('CTABar component should render', () => {
    const { container } = render(
      <CTABar>
        <Button variant="feature" color="light" leftIcon={<Phone />} ghost>
          Call Now: 888-888-8888
        </Button>
        <Button
          variant="feature"
          color="light"
          leftIcon={<QuestionAnswer />}
          ghost
        >
          Chat with agent
        </Button>
        <Button variant="feature" color="light" leftIcon={<Done />} ghost>
          Check Availability
        </Button>
      </CTABar>
    )

    expect(container).toMatchSnapshot()
  })

  test('CTABar with light prop should render', () => {
    const { container } = render(
      <CTABar light>
        <Button variant="feature" color="light" leftIcon={<Phone />} ghost>
          Call Now: 888-888-8888
        </Button>
        <Button
          variant="feature"
          color="light"
          leftIcon={<QuestionAnswer />}
          ghost
        >
          Chat with agent
        </Button>
        <Button variant="feature" color="light" leftIcon={<Done />} ghost>
          Check Availability
        </Button>
      </CTABar>
    )

    expect(container).toMatchSnapshot()
  })

  test('StickyBar component should render', () => {
    const { container } = render(
      <StickyBar showOriginal>
        <CTABar light>
          <Button variant="feature" color="light" leftIcon={<Phone />} ghost>
            Call Now: 888-888-8888
          </Button>
          <Button
            variant="feature"
            color="light"
            leftIcon={<QuestionAnswer />}
            ghost
          >
            Chat with agent
          </Button>
          <Button variant="feature" color="light" leftIcon={<Done />} ghost>
            Check Availability
          </Button>
        </CTABar>
      </StickyBar>
    )

    expect(container).toMatchSnapshot()
  })

  test('StickyBar component with isVisible prop should render', () => {
    const { container } = render(
      <StickyBar isVisible>
        <CTABar>
          <Button variant="feature" color="light" leftIcon={<Phone />} ghost>
            Call Now: 888-888-8888
          </Button>
          <Button
            variant="feature"
            color="light"
            leftIcon={<QuestionAnswer />}
            ghost
          >
            Chat with agent
          </Button>
          <Button variant="feature" color="light" leftIcon={<Done />} ghost>
            Check Availability
          </Button>
        </CTABar>
      </StickyBar>
    )

    expect(container).toMatchSnapshot()
  })

  test('standardImplementation component should render', () => {
    const { container } = render(<StandardImplementation />)

    expect(container).toMatchSnapshot()
  })
})
