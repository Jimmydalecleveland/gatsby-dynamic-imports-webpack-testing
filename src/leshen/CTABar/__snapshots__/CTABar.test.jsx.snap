// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTABar CTABar component should render 1`] = `
.emotion-12 {
  background-color: #0f6bc0;
  color: #fff;
  border-top: 5px solid;
  width: 100%;
  z-index: 3;
}

.emotion-12.original {
  border: 0;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px 16px;
  font: inherit;
  color: #000;
  background-color: #f4f5f6;
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-appearance: button;
  box-shadow: 0 0.3rem 1rem 0 rgba(0,0,0,0.1);
  color: #f4f5f6;
  box-shadow: none;
  background-color: transparent;
  padding-top: 12px;
  padding-bottom: 12px;
}

.emotion-2 > * {
  pointer-events: none;
}

.emotion-2 .e1u83bah1 svg,
.emotion-2 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  fill: #000;
  height: 16px;
  width: 16px;
}

.emotion-2 .button-text {
  margin-left: 8px;
}

.emotion-2:hover {
  color: #000;
  background-color: #e6e8eb;
}

.emotion-2:active {
  color: #000;
  background-color: #d8dcdf;
}

.emotion-2:disabled {
  background-color: #cacfd4;
  color: #5b636d;
}

.emotion-2 .e1u83bah1 svg,
.emotion-2 svg {
  fill: #f4f5f6;
}

.emotion-2:hover .e1u83bah1 svg,
.emotion-2:hover svg {
  fill: #000;
}

.emotion-2 .e1u83bah1 svg,
.emotion-2 svg {
  height: 24px;
  width: 24px;
}

.emotion-2 .button-text {
  margin-left: 12px;
}

.emotion-0 {
  font-size: 1.125rem;
  line-height: 1.5rem;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-0 {
    width: 100%;
  }
}

<div>
  <div
    class="emotion-12 emotion-13"
  >
    <button
      class="emotion-2 emotion-3"
      color="light"
      tabindex="0"
      type="button"
    >
      <div>
        Phone icon Mock
      </div>
      <span
        class="button-text emotion-0 emotion-1"
      >
        Call Now: 888-888-8888
      </span>
    </button>
    <button
      class="emotion-2 emotion-3"
      color="light"
      tabindex="0"
      type="button"
    >
      <div>
        QuestionAnswer icon Mock
      </div>
      <span
        class="button-text emotion-0 emotion-1"
      >
        Chat with agent
      </span>
    </button>
    <button
      class="emotion-2 emotion-3"
      color="light"
      tabindex="0"
      type="button"
    >
      <div>
        Done icon Mock
      </div>
      <span
        class="button-text emotion-0 emotion-1"
      >
        Check Availability
      </span>
    </button>
  </div>
</div>
`;

exports[`CTABar CTABar with light prop should render 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px 16px;
  font: inherit;
  color: #000;
  background-color: #f4f5f6;
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-appearance: button;
  box-shadow: 0 0.3rem 1rem 0 rgba(0,0,0,0.1);
  color: #f4f5f6;
  box-shadow: none;
  background-color: transparent;
  padding-top: 12px;
  padding-bottom: 12px;
}

.emotion-2 > * {
  pointer-events: none;
}

.emotion-2 .e1u83bah1 svg,
.emotion-2 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  fill: #000;
  height: 16px;
  width: 16px;
}

.emotion-2 .button-text {
  margin-left: 8px;
}

.emotion-2:hover {
  color: #000;
  background-color: #e6e8eb;
}

.emotion-2:active {
  color: #000;
  background-color: #d8dcdf;
}

.emotion-2:disabled {
  background-color: #cacfd4;
  color: #5b636d;
}

.emotion-2 .e1u83bah1 svg,
.emotion-2 svg {
  fill: #f4f5f6;
}

.emotion-2:hover .e1u83bah1 svg,
.emotion-2:hover svg {
  fill: #000;
}

.emotion-2 .e1u83bah1 svg,
.emotion-2 svg {
  height: 24px;
  width: 24px;
}

.emotion-2 .button-text {
  margin-left: 12px;
}

.emotion-0 {
  font-size: 1.125rem;
  line-height: 1.5rem;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-0 {
    width: 100%;
  }
}

.emotion-12 {
  background-color: #fff;
  color: #0f6bc0;
  border-top: 5px solid;
  width: 100%;
  z-index: 3;
}

.emotion-12.original {
  border: 0;
}

<div>
  <div
    class="emotion-12 emotion-13"
  >
    <button
      class="emotion-2 emotion-3"
      color="light"
      tabindex="0"
      type="button"
    >
      <div>
        Phone icon Mock
      </div>
      <span
        class="button-text emotion-0 emotion-1"
      >
        Call Now: 888-888-8888
      </span>
    </button>
    <button
      class="emotion-2 emotion-3"
      color="light"
      tabindex="0"
      type="button"
    >
      <div>
        QuestionAnswer icon Mock
      </div>
      <span
        class="button-text emotion-0 emotion-1"
      >
        Chat with agent
      </span>
    </button>
    <button
      class="emotion-2 emotion-3"
      color="light"
      tabindex="0"
      type="button"
    >
      <div>
        Done icon Mock
      </div>
      <span
        class="button-text emotion-0 emotion-1"
      >
        Check Availability
      </span>
    </button>
  </div>
</div>
`;

exports[`CTABar StickyBar component should render 1`] = `
.emotion-14 {
  background-color: #0f6bc0;
  color: #fff;
  border-top: 5px solid;
  width: 100%;
  z-index: 3;
}

.emotion-14.original {
  border: 0;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px 16px;
  font: inherit;
  color: #000;
  background-color: #f4f5f6;
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-appearance: button;
  box-shadow: 0 0.3rem 1rem 0 rgba(0,0,0,0.1);
  color: #f4f5f6;
  box-shadow: none;
  background-color: transparent;
  padding-top: 12px;
  padding-bottom: 12px;
}

.emotion-2 > * {
  pointer-events: none;
}

.emotion-2 .e1u83bah1 svg,
.emotion-2 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  fill: #000;
  height: 16px;
  width: 16px;
}

.emotion-2 .button-text {
  margin-left: 8px;
}

.emotion-2:hover {
  color: #000;
  background-color: #e6e8eb;
}

.emotion-2:active {
  color: #000;
  background-color: #d8dcdf;
}

.emotion-2:disabled {
  background-color: #cacfd4;
  color: #5b636d;
}

.emotion-2 .e1u83bah1 svg,
.emotion-2 svg {
  fill: #f4f5f6;
}

.emotion-2:hover .e1u83bah1 svg,
.emotion-2:hover svg {
  fill: #000;
}

.emotion-2 .e1u83bah1 svg,
.emotion-2 svg {
  height: 24px;
  width: 24px;
}

.emotion-2 .button-text {
  margin-left: 12px;
}

.emotion-0 {
  font-size: 1.125rem;
  line-height: 1.5rem;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-0 {
    width: 100%;
  }
}

.emotion-12 {
  background-color: #fff;
  color: #0f6bc0;
  border-top: 5px solid;
  width: 100%;
  z-index: 3;
}

.emotion-12.original {
  border: 0;
}

.emotion-32 {
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
  z-index: 3;
  position: fixed;
  opacity: 0;
  bottom: 0;
  width: 100%;
  -webkit-transition: -webkit-transform 0.5s cubic-bezier(1,0.05,0.17,1.09),opacity 0.5s cubic-bezier(1,0.05,0.17,1.09);
  -webkit-transition: transform 0.5s cubic-bezier(1,0.05,0.17,1.09),opacity 0.5s cubic-bezier(1,0.05,0.17,1.09);
  transition: transform 0.5s cubic-bezier(1,0.05,0.17,1.09),opacity 0.5s cubic-bezier(1,0.05,0.17,1.09);
  padding: 0;
}

<div>
  <div
    class="original emotion-14 emotion-13"
  >
    <div
      class="emotion-12 emotion-13"
    >
      <button
        class="emotion-2 emotion-3"
        color="light"
        tabindex="0"
        type="button"
      >
        <div>
          Phone icon Mock
        </div>
        <span
          class="button-text emotion-0 emotion-1"
        >
          Call Now: 888-888-8888
        </span>
      </button>
      <button
        class="emotion-2 emotion-3"
        color="light"
        tabindex="0"
        type="button"
      >
        <div>
          QuestionAnswer icon Mock
        </div>
        <span
          class="button-text emotion-0 emotion-1"
        >
          Chat with agent
        </span>
      </button>
      <button
        class="emotion-2 emotion-3"
        color="light"
        tabindex="0"
        type="button"
      >
        <div>
          Done icon Mock
        </div>
        <span
          class="button-text emotion-0 emotion-1"
        >
          Check Availability
        </span>
      </button>
    </div>
  </div>
  <div
    id="sticky-ref"
  />
  <div
    class="emotion-32 emotion-33"
    id="sticky-bar"
  >
    <div
      class="sticky emotion-14 emotion-13"
    >
      <div
        class="emotion-12 emotion-13"
      >
        <button
          class="emotion-2 emotion-3"
          color="light"
          tabindex="0"
          type="button"
        >
          <div>
            Phone icon Mock
          </div>
          <span
            class="button-text emotion-0 emotion-1"
          >
            Call Now: 888-888-8888
          </span>
        </button>
        <button
          class="emotion-2 emotion-3"
          color="light"
          tabindex="0"
          type="button"
        >
          <div>
            QuestionAnswer icon Mock
          </div>
          <span
            class="button-text emotion-0 emotion-1"
          >
            Chat with agent
          </span>
        </button>
        <button
          class="emotion-2 emotion-3"
          color="light"
          tabindex="0"
          type="button"
        >
          <div>
            Done icon Mock
          </div>
          <span
            class="button-text emotion-0 emotion-1"
          >
            Check Availability
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CTABar StickyBar component with isVisible prop should render 1`] = `
.emotion-12 {
  background-color: #0f6bc0;
  color: #fff;
  border-top: 5px solid;
  width: 100%;
  z-index: 3;
}

.emotion-12.original {
  border: 0;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px 16px;
  font: inherit;
  color: #000;
  background-color: #f4f5f6;
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-appearance: button;
  box-shadow: 0 0.3rem 1rem 0 rgba(0,0,0,0.1);
  color: #f4f5f6;
  box-shadow: none;
  background-color: transparent;
  padding-top: 12px;
  padding-bottom: 12px;
}

.emotion-2 > * {
  pointer-events: none;
}

.emotion-2 .e1u83bah1 svg,
.emotion-2 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  fill: #000;
  height: 16px;
  width: 16px;
}

.emotion-2 .button-text {
  margin-left: 8px;
}

.emotion-2:hover {
  color: #000;
  background-color: #e6e8eb;
}

.emotion-2:active {
  color: #000;
  background-color: #d8dcdf;
}

.emotion-2:disabled {
  background-color: #cacfd4;
  color: #5b636d;
}

.emotion-2 .e1u83bah1 svg,
.emotion-2 svg {
  fill: #f4f5f6;
}

.emotion-2:hover .e1u83bah1 svg,
.emotion-2:hover svg {
  fill: #000;
}

.emotion-2 .e1u83bah1 svg,
.emotion-2 svg {
  height: 24px;
  width: 24px;
}

.emotion-2 .button-text {
  margin-left: 12px;
}

.emotion-0 {
  font-size: 1.125rem;
  line-height: 1.5rem;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-0 {
    width: 100%;
  }
}

.emotion-16 {
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
  z-index: 3;
  position: fixed;
  opacity: 0;
  bottom: 0;
  width: 100%;
  -webkit-transition: -webkit-transform 0.5s cubic-bezier(1,0.05,0.17,1.09),opacity 0.5s cubic-bezier(1,0.05,0.17,1.09);
  -webkit-transition: transform 0.5s cubic-bezier(1,0.05,0.17,1.09),opacity 0.5s cubic-bezier(1,0.05,0.17,1.09);
  transition: transform 0.5s cubic-bezier(1,0.05,0.17,1.09),opacity 0.5s cubic-bezier(1,0.05,0.17,1.09);
  padding: 0;
}

<div>
  <div
    id="sticky-ref"
  />
  <div
    class="emotion-16 emotion-17"
    id="sticky-bar"
  >
    <div
      class="sticky emotion-12 emotion-13"
    >
      <div
        class="emotion-12 emotion-13"
      >
        <button
          class="emotion-2 emotion-3"
          color="light"
          tabindex="0"
          type="button"
        >
          <div>
            Phone icon Mock
          </div>
          <span
            class="button-text emotion-0 emotion-1"
          >
            Call Now: 888-888-8888
          </span>
        </button>
        <button
          class="emotion-2 emotion-3"
          color="light"
          tabindex="0"
          type="button"
        >
          <div>
            QuestionAnswer icon Mock
          </div>
          <span
            class="button-text emotion-0 emotion-1"
          >
            Chat with agent
          </span>
        </button>
        <button
          class="emotion-2 emotion-3"
          color="light"
          tabindex="0"
          type="button"
        >
          <div>
            Done icon Mock
          </div>
          <span
            class="button-text emotion-0 emotion-1"
          >
            Check Availability
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CTABar standardImplementation component should render 1`] = `
.emotion-16 {
  background-color: #0f6bc0;
  color: #fff;
  border-top: 5px solid;
  width: 100%;
  z-index: 3;
}

.emotion-16.original {
  border: 0;
}

.emotion-36 {
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
  z-index: 3;
  position: fixed;
  opacity: 0;
  bottom: 0;
  width: 100%;
  -webkit-transition: -webkit-transform 0.5s cubic-bezier(1,0.05,0.17,1.09),opacity 0.5s cubic-bezier(1,0.05,0.17,1.09);
  -webkit-transition: transform 0.5s cubic-bezier(1,0.05,0.17,1.09),opacity 0.5s cubic-bezier(1,0.05,0.17,1.09);
  transition: transform 0.5s cubic-bezier(1,0.05,0.17,1.09),opacity 0.5s cubic-bezier(1,0.05,0.17,1.09);
  padding: 0;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  margin: auto;
}

@media (min-width:768px) {
  .emotion-14 {
    padding: 8px;
  }
}

.emotion-5 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-5:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 8px 16px;
  font: inherit;
  color: #000;
  background-color: #f4f5f6;
  border: 0;
  border-radius: 4px;
  cursor: pointer;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  -webkit-appearance: button;
  box-shadow: 0 0.3rem 1rem 0 rgba(0,0,0,0.1);
  color: #f4f5f6;
  box-shadow: none;
  background-color: transparent;
  padding-top: 16px;
  padding-bottom: 16px;
  padding-left: 2.75rem;
  padding-right: 48px;
}

.emotion-3 > * {
  pointer-events: none;
}

.emotion-3 .e1u83bah1 svg,
.emotion-3 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  fill: #000;
  height: 16px;
  width: 16px;
}

.emotion-3 .button-text {
  margin-left: 8px;
}

.emotion-3:hover {
  color: #000;
  background-color: #e6e8eb;
}

.emotion-3:active {
  color: #000;
  background-color: #d8dcdf;
}

.emotion-3:disabled {
  background-color: #cacfd4;
  color: #5b636d;
}

.emotion-3 .e1u83bah1 svg,
.emotion-3 svg {
  fill: #f4f5f6;
}

.emotion-3:hover .e1u83bah1 svg,
.emotion-3:hover svg {
  fill: #000;
}

.emotion-3 .e1u83bah1 svg,
.emotion-3 svg {
  height: 32px;
  width: 32px;
}

.emotion-3 .button-text {
  margin-left: 16px;
}

@media (max-width:768px) {
  .emotion-3 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    padding: 8px 8px;
  }

  .emotion-3 .button-text {
    font-size: 14px;
    font-weight: normal;
    margin: 0;
  }
}

.emotion-0 {
  font-weight: 500;
  font-size: calc(0.5555555555555556vw + 17.333333333333332px);
  line-height: calc(1.1111111111111112vw + 18.666666666666668px);
}

@media (max-width:480px) {
  .emotion-0 {
    font-size: 20px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    font-size: 24px;
  }
}

@media (max-width:480px) {
  .emotion-0 {
    line-height: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    line-height: 32px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-0 {
    width: 100%;
  }
}

<div>
  <div
    class="original emotion-16 emotion-17"
  >
    <div
      class="emotion-14 emotion-15"
    >
      <a
        class="emotion-5 emotion-6"
        href="tel:1-555-555-5555"
        style="margin-right: 5vw;"
      >
        <button
          class="emotion-2 emotion-3 emotion-4"
          color="light"
          tabindex="0"
          type="button"
        >
          <div>
            Phone icon Mock
          </div>
          <span
            class="button-text emotion-0 emotion-1"
          >
            Call: 
            1-555-555-5555
          </span>
        </button>
      </a>
      <a
        class="emotion-5 emotion-6"
        href="/availability"
        style="margin-left: 5vw;"
      >
        <button
          class="emotion-2 emotion-3 emotion-4"
          color="light"
          tabindex="0"
          type="button"
        >
          <div>
            Done icon Mock
          </div>
          <span
            class="button-text emotion-0 emotion-1"
          >
            Check Availability
          </span>
        </button>
      </a>
    </div>
  </div>
  <div
    id="sticky-ref"
  />
  <div
    class="emotion-36 emotion-37"
    id="sticky-bar"
  >
    <div
      class="sticky emotion-16 emotion-17"
    >
      <div
        class="emotion-14 emotion-15"
      >
        <a
          class="emotion-5 emotion-6"
          href="tel:1-555-555-5555"
          style="margin-right: 5vw;"
        >
          <button
            class="emotion-2 emotion-3 emotion-4"
            color="light"
            tabindex="0"
            type="button"
          >
            <div>
              Phone icon Mock
            </div>
            <span
              class="button-text emotion-0 emotion-1"
            >
              Call: 
              1-555-555-5555
            </span>
          </button>
        </a>
        <a
          class="emotion-5 emotion-6"
          href="/availability"
          style="margin-left: 5vw;"
        >
          <button
            class="emotion-2 emotion-3 emotion-4"
            color="light"
            tabindex="0"
            type="button"
          >
            <div>
              Done icon Mock
            </div>
            <span
              class="button-text emotion-0 emotion-1"
            >
              Check Availability
            </span>
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
